exonum-java-proofs/src/test/java/com/exonum/binding/storage/proofs/map/flat/UncheckedFlatMapProofTest.java

```diff
@@ -141,60 +136,71 @@ public void mapProofWithEqualEntriesOrderShouldBeInvalid() {
+            Collections.singletonList(createBranchMapEntry(firstDbKey)),
+            Collections.singletonList(createLeafMapEntry(valueKey, FIRST_VALUE)),
+            Collections.singletonList(absentKey));
 
     CheckedMapProof checkedMapProof = uncheckedFlatMapProof.check();
     assertThat(checkedMapProof.getStatus(), equalTo(ProofStatus.CORRECT));
   }
 
   @Test
-  public void mapProofWithAbsentKeyShouldBeCorrect() {
-    byte[] firstKey = createPrefixed(bytes(0b001101), DbKey.KEY_SIZE);
-    byte[] valueKey = createPrefixed(bytes(0b011101), DbKey.KEY_SIZE);
-    byte[] absentKey = createPrefixed(bytes(0b111101), DbKey.KEY_SIZE);
+  public void mapProofWithSingleProofEntryShouldBeInvalid() {
+    DbKey firstDbKey = DbKeyTestUtils.branchKeyFromPrefix("1011111");
+    byte[] absentKey = DbKeyTestUtils.keyFromString("101111");
 
     UncheckedMapProof uncheckedFlatMapProof =
         new UncheckedFlatMapProof(
-            Collections.singletonList(createBranchMapEntry(firstKey)),
-            Collections.singletonList(createLeafMapEntry(valueKey, VALUE)),
-            Collections.singletonList(createAbsentLeafMapEntry(absentKey)));
+            Collections.singletonList(createBranchMapEntry(firstDbKey)),
+            Collections.emptyList(),
+            Collections.singletonList(absentKey));
 
     CheckedMapProof checkedMapProof = uncheckedFlatMapProof.check();
-    assertThat(checkedMapProof.getStatus(), equalTo(ProofStatus.INVALID_ORDER));
+    assertThat(checkedMapProof.getStatus(), equalTo(ProofStatus.NON_TERMINAL_NODE));
   }
 
-  private static MapProofEntry createBranchMapEntry(byte[] key) {
-    byte[] prefixedKey = createPrefixed(key, DbKey.KEY_SIZE);
-    return new MapProofEntry(
-        DbKey.newBranchKey(prefixedKey, key.length),
-        HashCode.fromBytes(key));
+  @Test
+  public void mapProofWithIncludedPrefixesShouldBeInvalid() {
+    DbKey firstDbKey = DbKeyTestUtils.branchKeyFromPrefix("11");
+    DbKey secondDbKey = DbKeyTestUtils.branchKeyFromPrefix("01");
+    byte[] absentKey = DbKeyTestUtils.keyFromString("111111");
+
+    UncheckedMapProof uncheckedFlatMapProof =
+        new UncheckedFlatMapProof(
+            Arrays.asList(
+                new MapProofEntry(firstDbKey, HashCode.fromBytes(firstDbKey.getKeySlice())),
+                new MapProofEntry(secondDbKey, HashCode.fromBytes(secondDbKey.getKeySlice()))),
+            Collections.emptyList(),
+            Collections.singletonList(absentKey));
+
+    CheckedMapProof checkedMapProof = uncheckedFlatMapProof.check();
+    assertThat(checkedMapProof.getStatus(), equalTo(ProofStatus.INVALID_STRUCTURE));
   }
 
-  private static MapProofEntryLeaf createLeafMapEntry(byte[] key, byte[] value) {
-    byte[] prefixedKey = createPrefixed(key, DbKey.KEY_SIZE);
-    return new MapProofEntryLeaf(DbKey.newLeafKey(prefixedKey), value);
+  private static MapProofEntry createBranchMapEntry(DbKey key) {
```

Comment:
proof _might_ contain proof nodes, corresponding to leaves, plus this method does not have branch-specific logic.
I'd suggest `createProofEntry(DbKey key)` or `createMapProofEntry(DbKey key)`